docs/beyond_basics/101-168-dvc.rst

```diff
@@ -0,0 +1,1032 @@
+       train_data, labels = load_data(train_csv_path)
+       rf = RandomForestClassifier()
+       trained_model = rf.fit(train_data, labels)
+       dump(trained_model, repo_path / "model/model.joblib")
+
+   if __name__ == "__main__":
+       repo_path = Path(__file__).parent.parent
+       main(repo_path)
+   EOT
+
+We need to save this change:
+
+.. runrecord:: _examples/DL-101-168-177
+   :workdir: DVCvsDL/DVC-DataLad
+   :language: console
+
+   $ datalad save -m "Switch to random forrest classification" code/train.py
+
+Afterwards, we can rerun all run records between the tags ``ready-for-analysis`` and ``SGD`` using :command:`datalad rerun`.
+We could automatically compute this on a different branch if we wanted to by using the ``branch`` option:
+
+.. runrecord:: _examples/DL-101-168-178
+   :workdir: DVCvsDL/DVC-DataLad
+   :language: console
+
+   $ datalad rerun --branch="randomforrest" -m "Recompute classification with random forrest classifier" ready-for-analysis..SGD
+
+Done!
+The difference in accuracies between models could now for example be compared with a ``git diff``:
+
+
+.. runrecord:: _examples/DL-101-168-179
+   :workdir: DVCvsDL/DVC-DataLad
+   :language: console
+
+   $ git diff SGD -- metrics/accuracy.json
+
+Even though there is no one-to-one correspondence between a DVC and a DataLad workflow, a DVC workflow can also be implemented with DataLad.
+
+.. only:: adminmode
+
+   We need to clean up -- reset the state of the "data version control" repo to its original state, force push
+
+   .. runrecord:: _examples/DL-101-168-190
+      :workdir: DVCvsDL/DVC
+      :language: console
+
+      $ git checkout master
+      $ git reset --hard b796ba195447268ebc51e20a778fb2db9f11e341
+      $ git push --force origin master
+      # delete the branches and tags
+      $ git push origin :random_forrest :sgd-pipeline
+      $ git tag -d random-forrest sgd-pipeline
+
+Summary
+^^^^^^^
+
+DataLad and DVC aim to solve the same problems: Version control data, sharing data, and enabling reproducible analyses.
+DataLad provides generic solutions to these issues, while DVC is tuned for machine-learning pipelines.
+Despite their similar purpose, the looks, feels and functions of both tools are different, and its a personal decision which one you feel more comfortable with.
+Using DVS requires solid knowledge of Git, because DVC workflows heavily rely on effective Git practices, such as branching, tags, and ``.gitignore`` files.
+But despite the reliance on Git, DVC barely integrates with Git -- changes done to files in DVC can not be detected by Git and vice versa, DVC and Git aspects of a repository have to be handled in parallel by the user, and DVC and Git have distinct command functions and concepts that nevertheless share the same name.
+Thus, DVC users need to master Git *and* DVS workflows and intertwine them correctly.
+In return, DVC provides users with workflow management and reporting tuned to machine learning analyses.
```

Comment:
```suggestion
In return, DVC provides users with workflow management and reporting tuned to machine learning analyses. It also provides a somewhat more lightweight and uniform across operating and file systems approach to "data version control" than git-annex used by DataLad.
```
I think there should be some statement on that.  With `git-annex` we have at times a VERY complex and heterogeneous across systems setup (all the adjusted branches, git link vs symlink, etc).  I think the advantage of DVC in keeping data management very simple (although not as nicely integrated with git) should be mentioned.

Reply:
I wonder if there is a place (or may be already a reference) which would demonstrate pros of "symlinked" setup of git-annex:  `git checkout` is **very fast** regardless of the sizes of data files etc, and you get your repository in a consistent state with data files of needed versions etc.  How is it (get another consistent state) to be done with `dvc`?

Reply:
> I wonder if there is a place (or may be already a reference) which would demonstrate pros of "symlinked" setup of git-annex: `git checkout` is **very fast** regardless of the sizes of data files etc, and you get your repository in a consistent state with data files of needed versions etc. How is it (get another consistent state) to be done with `dvc`?

I have footnoted the section on how the object tree works in the section on version controlling data, but haven't explicitly stated such advantages in the text of this section - sounded like "showing of". We could do it down here, maybe?